### DISCUSSION TITLE
How to properly extend InputBase props?

### DISCUSSION THREAD

**Discussion by jhegele** [mantinedev/mantine#5546] [Q&A] [answered]
I'm trying to build a generic component that I can use to add input masking on text inputs. I can't seem to get the Typescript props right, though. Here's how I am currently defining my component:

```TSX
import { InputBase, type InputBaseProps } from '@mantine/core'
import { forwardRef } from 'react'
import { IMaskInput } from 'react-imask'

export interface MaskedTextInputProps extends InputBaseProps {
    mask: string
}

export const MaskedTextInput = forwardRef<
    HTMLInputElement,
    MaskedTextInputProps
>(({ mask, ...inputProps }, ref) => (
    <InputBase
        {...inputProps}
        ref={ref}
        component={IMaskInput}
        mask={mask}
        type="text"
    />
))

MaskedTextInput.displayName = 'MaskedTextInput'
```

The problem that I'm running into here is that it seems like `InputBaseProps` is not the complete set of props supported by the `<InputBase>` element. For example, as defined, when I import this component, Typescript complains that there is no `onChange` prop.

I suspect I'm missing something here but I've been through the docs and just can't figure out how to actually extend the props for the `<InputBase>` element. Appreciate any help!

**Comment by yverby** [comment, marked as answer]
Following [this](https://mantine.dev/guides/polymorphic/#wrapping-polymorphic-components) guide you can try

```ts
export interface MaskedTextInputProps
  extends InputBaseProps,
    Omit<
      React.ComponentPropsWithoutRef<typeof IMaskInput>,
      keyof InputBaseProps
    > {}
```

**Reply by jhegele** [reply]
Thanks! That seems to have done the trick.
